spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskSchedulerController.java

```diff
@@ -69,16 +70,16 @@ public TaskSchedulerController(SchedulerService schedulerService) {
 	/**
 	 * Return a page-able list of {@link ScheduleInfo}s.
 	 *
-	 * @param pageable  page-able collection of {@code ScheduleResource}.
 	 * @param assembler assembler for the {@link ScheduleInfo}
 	 * @return a list of Schedules
 	 */
 	@RequestMapping(value = "", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public PagedResources<ScheduleInfoResource> list(Pageable pageable,
+	public PagedResources<ScheduleInfoResource> list(
 			PagedResourcesAssembler<ScheduleInfo> assembler) {
+		List<ScheduleInfo> result = this.schedulerService.list();
+		Pageable pageable = new PageRequest(0, result.size());
```

Comment:
This way we enforce the pageable content size based on the result. I think this has to come from the client; otherwise, the client implementations (ex: web UI) wouldn't be able to request the desired page.